pkg/workloads/cortex/tf_api/api.py

```diff
@@ -385,14 +404,16 @@ def start(args):
                 sys.exit(1)
 
         time.sleep(5)
-    logger.info(
-        "model_signature: {}".format(
-            extract_signature(
-                local_cache["metadata"]["signatureDef"],
-                local_cache["api"]["tf_serving"]["signature_key"],
-            )
-        )
-    )
+
+    signature_key = None
+    if api.get("tf_serving") is not None and api["tf_serving"].get("signature_key") is not None:
+        signature_key = api["tf_serving"]["signature_key"]
+
+    key, metadata = extract_signature(local_cache["metadata"]["signatureDef"], signature_key)
+
+    local_cache["signature_key"] = key
+    local_cache["signature"] = metadata
```

Comment:
What is the relation between `local_cache[“signature”]` (which is set to `metadata`) and `local_cache[“metadata”]`, and should we rename one or both of them?